packages/astro/src/runtime/server/serialize.ts

```diff
@@ -67,6 +70,15 @@ function convertToSerializedForm(
 		case '[object Array]': {
 			return [PROP_TYPE.JSON, JSON.stringify(serializeArray(value, metadata))];
 		}
+		case '[object Uint8Array]': {
```

Comment:
you need to pass `metadata` through to these functions

Reply:
I've added the value passing